Sources/Orbit/Components/Slider.swift

```diff
@@ -0,0 +1,421 @@
+        let normalisied = min(sliderWidth, max(0, position))
+        return normalisied
+    }
+
+    public init(
+        value: Binding<Value>,
+        valueLabel: String,
+        endLabel: String,
+        bounds: ClosedRange<Value>,
+        step: Value.Stride? = nil
+    ) {
+        self.value = .single(value)
+        self.valueLabel = valueLabel
+        self.startLabel = ""
+        self.endLabel = endLabel
+
+        self.bounds = bounds
+        self.step = step
+
+        self._firstValue = State(initialValue: value.wrappedValue)
+        buckets = bounds.chunked(into: step ?? Value.Stride(bounds.upperBound))
+    }
+
+    public init(
+        selectedRange: Binding<ClosedRange<Value>>,
+        valueLabel: String,
+        startLabel: String,
+        endLabel: String,
+        bounds: ClosedRange<Value>,
+        step: Value.Stride? = nil
+    ) {
+        assert(step != 0, "step can'be 0")
+
+        self.value = .range(selectedRange)
+        self.valueLabel = valueLabel
+        self.startLabel = startLabel
+        self.endLabel = endLabel
+        self.bounds = bounds
+        self.step = step
+
+        self._firstValue = State(initialValue: selectedRange.wrappedValue.lowerBound)
+        self._secondValue = State(initialValue: selectedRange.wrappedValue.upperBound)
+
+        self.buckets = bounds.chunked(into: step ?? Value.Stride(bounds.upperBound))
+    }
+}
+
+struct SliderThumb: View {
+
+    @Environment(\.sizeCategory) var sizeCategory
+
+    public static let circleDiameter: CGFloat = 24
+    public static let dotDiameter: CGFloat = 8
+
+    public var body: some View {
+        Circle()
+            .frame(width: circleDiameter, height: circleDiameter)
+            .foregroundColor(Color.white)
+            .elevation(.level1)
+            .overlay(indicatorSymbol)
+    }
+
+    @ViewBuilder var indicatorSymbol: some View {
+        Circle()
+            .foregroundColor(Color.blueNormal)
+            .frame(width: dotDiameter, height: dotDiameter)
+    }
+
+    var circleDiameter: CGFloat {
+        Self.circleDiameter * sizeCategory.controlRatio
+    }
+
+    var dotDiameter: CGFloat {
+        Self.dotDiameter * sizeCategory.ratio
+    }
+}
+
+extension ClosedRange where Bound: BinaryFloatingPoint {
+    func chunked(into size: Bound.Stride) -> [Range<Bound>] {
+        stride(from: lowerBound, to: upperBound, by: size).map { value in
+            value ..< Swift.min(value.advanced(by: size), upperBound)
+        }
+    }
+}
+
+extension Range where Bound: BinaryFloatingPoint {
+    var middle: Bound {
+        lowerBound + (upperBound - lowerBound) / 2
+    }
+}
+
+extension Slider where Value == Double {
+    init(value: Binding<Value>, bounds: ClosedRange<Value>, step: Value.Stride? = nil) {
+        self.init(
+            value: value,
+            valueLabel: value.wrappedValue.formatted,
+            endLabel: bounds.upperBound.formatted,
```

Comment:
I wonder what this `formatted` does in various locales 🤔

Reply:
We should avoid formatting it at Orbit level. The Orbit should take exactly what the call size wants to display, 1:1

Reply:
It's the `internal init` visible and used only in  Previews, so it won't affect users

Reply:
But those previews can be used as examples by devs. We should not use internal convenience inits in previews, but the real API, so that we can even test that the API is easy to use.